### ISSUE TITLE
Voltages from two polarizations should not be summed

### ISSUE THREAD

**Issue by david-macmahon** [lacker/seticore#22]
It looks like the `beamform_voltage` function in [viewer.py](https://github.com/lacker/seticore/blob/27971502592eb761e3fe2e09d0dd79243dd449c4/python/viewer.py#L248) sums the polarization voltages together.  The polarizations should not be summed in voltage, but only in power.

**Comment by lacker** [comment]
When I first saw this comment, my thoughts were, oh no every beamforming this code has ever done was wrong. But fortunately the c++ beamforming logic does appear to be combining the polarizations in power.

https://github.com/lacker/seticore/blob/master/beamformer.cu#L237

Whew. I thought I had validated the python beamforming logic by comparing to the c++ output but there is no automated testing for it so I'll have to take a closer look at that.

**Comment by lacker** [comment]
When I rerun the validation logic it does in fact report that c++ and python beamforming do not match up. Maybe I broke it at some point without noticing... I should make some python unit tests here.

```
original power: 1.094615e+17
recalculated power: 8.274025e+16
```

**Comment by lacker** [comment]
Fixed in https://github.com/lacker/seticore/commit/dccb5599bffa1a6f3b66bb0b43fb2879cb2f8ecf

The values still seem off by a little bit but maybe this is within the allowable realm of GPU multiplication error.

```
original power: 1.094615e+17
recalculated power: 1.095399e+17
```

**Comment by david-macmahon** [comment]
So is the fix just to make `beamform_voltage` return a 3D array rather than a 2D array (i.e. with an extra polarization dimension) by excluding the polarization dimension from the `sum()` call?  And maybe a corresponding change (to sum polarizations) in the `beamform_power` function???

**Comment by lacker** [comment]
Yep

**Comment by david-macmahon** [comment]
Yeah, it's not just multiplication error, but also the non-deterministic order of operations (especially summing) that can lead to slightly different results.
